src/inspector_io.cc

```diff
@@ -363,12 +391,18 @@ void InspectorIo::DispatchMessages() {
 
 // static
 void InspectorIo::MainThreadAsyncCb(uv_async_t* req) {
-  InspectorIo* io = node::ContainerOf(&InspectorIo::main_thread_req_, req);
-  io->DispatchMessages();
+  async_agent* pair = node::ContainerOf(&async_agent::first, req);
+  // Note that this may be called after io was closed or even after a new
+  // one was created and ran.
+  InspectorIo* io = pair->second->io();
+  if (io != nullptr)
```

Comment:
do whatever is node style, but since pointers evaluate to true/false, why not just `if(!io)`?